src/cdef.rs

```diff
@@ -559,67 +657,41 @@ pub fn cdef_filter_frame<T: Pixel>(
   };
 
   for p in 0..3 {
-    let rec_w = rec.planes[p].cfg.width;
-    let rec_h = rec.planes[p].cfg.height;
+    let rec_w = rec.planes[p].rect().width;
+    let rec_h = rec.planes[p].rect().height;
     let mut cdef_region = cdef_frame.planes[p].region_mut(Area::Rect {
       x: -2,
       y: -2,
       width: rec_w + 4,
       height: rec_h + 4,
     });
+
+    let in_padded_region = in_padded_frame.planes[p].region(Area::Rect {
+      x: rec.planes[p].rect().x - 2,
+      y: rec.planes[p].rect().y - 2,
+      width: rec_w + 4,
+      height: rec_h + 4,
+    });
+
     for row in 0..cdef_region.rect().height {
-      // pad first two elements of current row
-      {
-        let cdef_row = &mut cdef_region[row][..2];
-        cdef_row[0] = CDEF_VERY_LARGE;
-        cdef_row[1] = CDEF_VERY_LARGE;
-      }
-      // pad out end of current row
-      {
-        let cdef_row = &mut cdef_region[row][rec_w + 2..];
-        for x in cdef_row {
-          *x = CDEF_VERY_LARGE;
-        }
-      }
-      // copy current row from rec if we're in data, or pad if we're in first two rows/last N rows
-      {
-        let cdef_row = &mut cdef_region[row][2..rec_w + 2];
-        if row < 2 || row >= rec_h + 2 {
-          for x in cdef_row {
-            *x = CDEF_VERY_LARGE;
-          }
-        } else {
-          let rec_stride = rec.planes[p].cfg.stride;
-          for (x, y) in cdef_row.iter_mut().zip(
-            rec.planes[p].data_origin()
-              [(row - 2) * rec_stride..(row - 1) * rec_stride]
-              .iter(),
-          ) {
-            *x = u16::cast_from(*y);
-          }
-        }
-      }
```

Comment:
Oh wow, don't know how I missed this extra padding step as having become redundant.  Nice catch.  Maaaaybe... break this out as a separate change that happens first?  Just in case?

Reply:
Do you mean separating this change for padding as different PR?

Reply:
It need not be a separate PR, just a separate commit.  

Looking at it carefully, I think you're right to remove the padding code.  However, my intuition tells me there is a possibility I could be wrong.  To make future maintenance easier in that case (even if I'm the one who has to debug it), being able to git bisect this change separately would be very useful.

Reply:
I make it to separate commit.